src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1221,30 +1227,29 @@ impl<T: 'static> EventProcessor<T> {
     }
 
     fn handle_pressed_keys<F>(
-        &self,
+        wt: &super::EventLoopWindowTarget<T>,
         window_id: crate::window::WindowId,
         state: ElementState,
+        mod_keymap: &ModifierKeymap,
+        device_mod_state: &mut ModifierKeyState,
         callback: &mut F,
     ) where
         F: FnMut(Event<'_, T>),
     {
-        let wt = get_xtarget(&self.target);
-
         let device_id = mkdid(util::VIRTUAL_CORE_KEYBOARD);
-        let modifiers = self.device_mod_state.modifiers();
-
-        // Get the set of keys currently pressed and apply Key events to each
-        let keys = wt.xconn.query_keymap();
-
-        for keycode in &keys {
-            if keycode < 8 {
-                continue;
-            }
+        let modifiers = device_mod_state.modifiers();
 
+        // Get the set of keys currently pressed and apply Key events to each and also updating
+        // the pressed modifiers.
+        for keycode in wt.xconn.query_keymap().into_iter().filter(|k| *k >= 8) {
```

Comment:
This isn't different from the existing code, but I'd like to know why keycodes below 8 are filtered? Just seems kinda arbitrary without documentation.

Reply:
Running `xmodmap -pk` gives me a somewhat unsatisfying answer:

```
There are 7 KeySyms per KeyCode; KeyCodes range from 8 to 255.

    KeyCode	Keysym (Keysym)	...
    Value  	Value   (Name) 	...

      8    	
      9    	0xff1b (Escape)	0x0000 (NoSymbol)	0xff1b (Escape)	
...
```

Reply:
"Keycodes lie in the inclusive range [8,255]" [[1]]( https://www.x.org/releases/X11R7.7/doc/xproto/x11protocol.html#Keyboards)

Reply:
Here's a potential explanation: http://who-t.blogspot.com/2014/12/why-255-keycode-limit-in-x-is-real.html?showComment=1417626693719#c1258137695853791876